src/java.base/share/classes/jdk/internal/foreign/abi/NativeEntryPoint.java

```diff
@@ -48,7 +48,29 @@ public class NativeEntryPoint {
     private record CacheKey(MethodType methodType, ABIDescriptor abi,
                             List<VMStorage> argMoves, List<VMStorage> retMoves,
                             boolean needsReturnBuffer, int capturedStateMask,
-                            boolean needsTransition) {}
+                            boolean needsTransition) {
+
+        @Override
+        public boolean equals(Object o) {
+            if (!(o instanceof CacheKey other)) return false;
+
```

Comment:
Might be worth checking `o == this`

Reply:
Always used with `new CacheKey` in `NativeEntryPoint.make`, no benefit too.